### ISSUE TITLE
Serialization of H256 fails

### ISSUE THREAD

**Issue by niklasad1** [tomusdrw/rust-web3#163]
```rustc 
let h: H256 = hex!("a17b45b011dad358740ccb65383e0900b086643d15192060f3983160492cca15").into();
web3.eth().transaction_receipt(h).wait().unwrap(); 
```

#### Error
```bash
eth_getTransactionReceipt: 
Err(Error(Decoder("Error(\"invalid type: null, 
expected a 0x-prefixed hex string with length between (0; 64]\", line: 0, column: 0)"), 
State { next_error: None, backtrace: InternalBacktrace { backtrace: None } })
```

**Comment by drewstone** [comment]
I am also getting errors similar when I try to deploy a contract based on the example:
```
Err(Error(Api(Decoder("Error(\"invalid type: integer `1`, expected a 0x-prefixed hex string with length between (0; 16]\", line: 0, column: 0)")), State { next_error: None, backtrace: InternalBacktrace { backtrace: None } }))
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: Error(Api(Decoder("Error(\"invalid type: integer `1`, expected a 0x-prefixed hex string with length between (0; 16]\", line: 0, column: 0)")), State { next_error: None, backtrace: InternalBacktrace { backtrace: None } })', libcore/result.rs:1009:5
```

**Comment by tomusdrw** [comment]
@niklasad1 what version did you use? I think `null`s in receipts has been fixed in #164 

@drewstone what RPC backend do you use (geth, parity, infura)? What RPC are you calling? It seems that the response contains a raw integer `1`, while the library expects a hex-encoded value.

**Comment by drewstone** [comment]
I’m using test rpc as a backend (I guess more formally known as ganache?)

**Comment by tomusdrw** [comment]
I see, I remember some issues with the return types, but I hoped they were all fixed. What RPC method are you calling?

**Comment by drewstone** [comment]
So I’m trying to create a contract which I believe works from looking at
the logs of testrpc, however the program crashed or stalled and didn’t
return the contract object.

Unfortunately, my computer also just broke so I am in a bit of a tough spot
remembering exactly.

**Comment by CPerezz** [comment]
> I am also getting errors similar when I try to deploy a contract based on the example:
> 
> ```
> Err(Error(Api(Decoder("Error(\"invalid type: integer `1`, expected a 0x-prefixed hex string with length between (0; 16]\", line: 0, column: 0)")), State { next_error: None, backtrace: InternalBacktrace { backtrace: None } }))
> thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: Error(Api(Decoder("Error(\"invalid type: integer `1`, expected a 0x-prefixed hex string with length between (0; 16]\", line: 0, column: 0)")), State { next_error: None, backtrace: InternalBacktrace { backtrace: None } })', libcore/result.rs:1009:5
> ```

Seems by reviewing the error that you are trying to store an Address on an Integer variable. That's why **unwrap** it's failing. 
Just like on the examples of the repo, try to do something like:
```rust
let contract_addr: web3::types::Address = 
......
Deploy method call
.......
```

Due to the recent changes on the ethereum_types, maybe you can do something like:

```rust
let contract_addr: ethereum_types::H160 = 
.....
Deploy method call..
.....
```

Can you provice RPC traces?

**Comment by niklasad1** [comment]
Fixed in version `0.6`
